### PR TITLE
feat: default db check for postgres and redshift

### PR DESCRIPTION
**Pull request by Sanix-Darker**
## WHAT

- added default db for postgres
- added default db for redshift
- added a new status check for request on default database
- adapt code & tests accordingly to changes for postgres and redshift

### CODE REVIEW

File: toucan_connectors/redshift/redshift_database_connector.py [hunk 1]
```diff
@@ -312,6 +322,15 @@ def get_status(self) -> ConnectorStatus:
...
             return ConnectorStatus(status=False, details=self._get_details(1, False), error=str(e))
+
+        # Basic db query
+        try:
+            self._retrieve_tables(database=self.default_database)
+        except Exception as e:
+            return ConnectorStatus(
+                status=False, details=self._get_details(3, False), error=e.args[0]
+            )
+
```
Comment (jGundermann) [inline toucan_connectors/redshift/redshift_database_connector.py:333]: Maybe we should do a simple connect like postgres connector here : 

```suggestion

        # Basic db query
        try:
            redshift_connector.connect(**self.get_connection_params(database=database_name))
        except Exception as e:
            return ConnectorStatus(
                status=False, details=self._get_details(3, False), error=e.args[0]
            )

```

Comment (jGundermann) [reply toucan_connectors/redshift/redshift_database_connector.py:333]: we have to implement get_connection_params accordignly for redshift

Comment (Sanix-Darker) [reply toucan_connectors/redshift/redshift_database_connector.py]: Did you mean something like that : https://github.com/ToucanToco/toucan-connectors/pull/616/commits/0aa0d96c0651a6aae818b73e61e67744c6c7e146 ?

File: toucan_connectors/postgres/postgresql_connector.py [hunk 2]
```diff
@@ -166,7 +178,17 @@ def get_status(self) -> ConnectorStatus:
                     status=False, details=self._get_details(3, False), error=e.args[0]
                 )
 
-        return ConnectorStatus(status=True, details=self._get_details(3, True), error=None)
+        # Basic db query
+        try:
+            connection = pgsql.connect(**self.get_connection_params(database=self.default_database))
+            with connection.cursor() as cursor:
+                cursor.execute("""select 1;""")
```
Comment (raphaelvignes) [inline toucan_connectors/postgres/postgresql_connector.py:185]: :+1:

File: toucan_connectors/postgres/postgresql_connector.py [hunk 3]
```diff
@@ -166,7 +178,15 @@ def get_status(self) -> ConnectorStatus:
...
-        return ConnectorStatus(status=True, details=self._get_details(3, True), error=None)
+        # Basic db query
+        try:
+            self._list_tables_info(database_name=self.default_database)
+        except (Exception, pgsql.Error) as e:
+            return ConnectorStatus(
+                status=False, details=self._get_details(4, False), error=e.args[0]
+            )
+
+        return ConnectorStatus(status=True, details=self._get_details(4, True), error=None)
```
[the remark this answers is not in the document]
Comment (Sanix-Darker) [reply toucan_connectors/postgres/postgresql_connector.py]: https://github.com/ToucanToco/toucan-connectors/pull/616/commits/32d5c709672269a88bbb01d472258f222aaa86fe

**Review by jGundermann — APPROVED** [review]
LGTM

File: toucan_connectors/postgres/postgresql_connector.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (raphaelvignes) [reply toucan_connectors/postgres/postgresql_connector.py]: :+1:

**Review by raphaelvignes — APPROVED** [review]
Thx

**Pull request merged**